### PR TITLE
[new Model] VespaG

### PR DESCRIPTION
**Pull request by JSchlensok**
We'd love to contribute [VespaG](https://github.com/jSchlensok/vespag) as a new baseline model for the `DMS substitutions` benchmark.
You can reproduce our results as follows:
- download pre-computed ESM2 embeddings from [Zenodo](https://zenodo.org/records/11085958/files/proteingym_217_esm2.h5?download=1)
- create & activate Conda environment from `proteingym/baselines/vespag/environment.yml`
- run `scripts/scoring_DMS_zero_shot/scoring_VespaG_substitutions.sh`

Pre-computed Spearman values are also available on [Zenodo](https://zenodo.org/records/11085958/files/proteingym_performance_vespag.csv?download=1), as are [raw predicted scores for all mutants](https://zenodo.org/records/11085958/files/vespag_proteingym_rawpreds_by_training_dataset.zip?download=1) (training dataset used for final method: Human).

VespaG achieves an average Spearman r of 0.483 overall and a weighted average of 0.459.

Let me know if anything is missing or unclear!

### CODE REVIEW

**Review by aaronkollasch — CHANGES_REQUESTED** [review]
Hi @JSchlensok,

Thanks for submitting VespaG to ProteinGym, nice work! Sorry for the delay in reviewing.

We have a few suggested comments and suggested changes to make before merging. Replication looks good so far on our end, but I still have some checks to do for that as well.

Cheers,
Aaron

File: proteingym/baselines/vespag/environment.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      - rich
+      - tqdm
+      - transformers
+      - typing_extensions
+      - wandb
+  - funcy=1.17
+  - h5py
+  - joblib
+  - MKL=2024.0
+  - numpy
```
[the remark this answers is not in the document]
Comment (JSchlensok) [reply proteingym/baselines/vespag/environment.yml:23]: This has been resolved in https://github.com/JSchlensok/VespaG/commit/dc16036c59060815d77efcd6e5385ad0288da823, will pull later

File: proteingym/baselines/vespag/model_weights/state_dict_v1.pt
[the remark this answers is not in the document]
Comment (JSchlensok) [reply proteingym/baselines/vespag/model_weights/state_dict_v1.pt:1]: Yes, that is certainly possible.

[the remark this answers is not in the document]
Comment (JSchlensok) [reply proteingym/baselines/vespag/model_weights/state_dict_v1.pt:1]: resolved it via Torch Hub from the upstream GitHub repo instead

**Comment by JSchlensok** [comment]
Hi @aaronkollasch,
thanks for taking the time and don't worry about the delay.
Took me a bit to get back to you as well, sorry! All requested changes have been addressed now. Let me know if you need anything further!

Best,
Julius

**Review by aaronkollasch — APPROVED** [review]

**Comment by pascalnotin** [comment]
Thank you for the PR @JSchlensok, and for the review @aaronkollasch -- merging to main!

**Pull request merged**